src/main/java/net/sf/jabref/gui/JabRefFrame.java

```diff
@@ -886,12 +886,23 @@ public boolean quit() {
             for (int i = 0; i < tabbedPane.getTabCount(); i++) {
                 if (baseAt(i).isBaseChanged()) {
                     tabbedPane.setSelectedIndex(i);
-                    int answer = JOptionPane.showConfirmDialog
-                            (JabRefFrame.this, Localization.lang
-                                            ("Database has changed. Do you "
-                                                    + "want to save before closing?"),
-                                    Localization.lang("Save before closing"),
-                                    JOptionPane.YES_NO_CANCEL_OPTION);
+                    Object[] options = {Localization.lang("Save changes"),
+                            Localization.lang("Discard changes"),
+                            Localization.lang("Return to JabRef")};
+                    String filename;
+
+                    if (baseAt(i).getFile() != null) {
+                        filename = baseAt(i).getFile().getAbsolutePath();
+                    } else {
+                        filename = GUIGlobals.untitledTitle;
+                    }
+                    int answer = JOptionPane.showOptionDialog(JabRefFrame.this,
+                            "<html><br><b>" + Localization.lang("Database %0 has changed.", filename) + "</b>\n\n"
+                                    + Localization.lang(
+                                            "Click \"%0\" to save the changes,\nclick \"%1\" to discard any changes,\nclick \"%2\" to return to the main window.",
```

Comment:
I think we can skip this long localization as the different options are self-explanatory. Just state that the Database %0 has changed.